src/MixedModeBroadcastAD.jl

```diff
@@ -13,48 +13,41 @@ using CUDAnative
 # CUDA utilities #
 ##################
 
-cuda_dimensions(a::AbstractArray) = cuda_dimensions(length(a))
-
-function cuda_dimensions(n::Integer)
+@inline function cuda_dimensions(n::Integer)
     threads = min(n, 256)
-    ceil(Int, n / threads), threads
+    return ceil(Int, n / threads), threads
 end
 
-macro cuda_linear_index(A)
-    esc(quote
-        i = (blockIdx().x-UInt32(1)) * blockDim().x + threadIdx().x
-        i > length($A) && return
-        i
-    end)
+@inline function cuda_linear_index(n::Integer)
+    i = (blockIdx().x - UInt32(1)) * blockDim().x + threadIdx().x
+    i > n && return
```

Comment:
This needs to be a macro to cause an exit of the parent function (a design decision from the pre-bitstype union era, we could probably get away returning `nothing`).

Reply:
Oh, okay. Thanks.